src/main/java/dataloader/CsvDataLoader.java

```diff
@@ -0,0 +1,46 @@
+package dataloader;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import tech.tablesaw.api.Table;
+import tech.tablesaw.io.csv.CsvReadOptions;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.TreeMap;
+
+
+public class CsvDataLoader {
+
+    /**
+     * Simple class that parses data set from a csv file
+     * without knowing the headers of the file
+     */
+    public void parseFile(String csvFile, TreeMap dataSet) throws IOException {
+
+        Reader csvData = new FileReader(csvFile);
+
+        CSVParser parser = CSVParser.parse(csvData, CSVFormat.DEFAULT);
+
+        for (CSVRecord record : parser) {
+            for (String field : record) {
+                dataSet.keySet();
+                dataSet.get(field);
+
+                System.out.println("\"" + field + "\", ");
+            }
+        }
+    }
+
+    /**
+     * Reads from csv in file system with columns separated by commas
+     * and the file has a header row.
+     * Missing values are treated with an indicator.
+     */
+    public void parseFile(String csvFile, Table dataSet) throws IOException {
+        dataSet.read().usingOptions(CsvReadOptions.builder(csvFile).missingValueIndicator("-"));
```

Comment:
is the ```read``` method a static method? If ```Table``` cannot be instantiated per se we have two options: either the method is static so we should write ```Table.read()......``` or instantiate a concrete subclass of this class

Reply:
from what I can make of `tablesaw` documentation, the `read()` method is static.

Reply:
```Table t = Table.read().file("myFile.csv");``` you want sth like this. I took it from here: https://jtablesaw.github.io/tablesaw/userguide/importing_data.html

Reply:
yes, I've seen the reference. but I've decided to treat missing values here, so the documentation of `tablesaw`, favours the implementation I've used. check out the new implementation and tell me what you think.

Reply:
the test now hangs

Reply:
I don't get you :) :)

Reply:
The way you changed the parseFile in the CsvDataLoader, you don't get the Table back you need to return it